DatadogSessionReplay/Sources/Recorder/RecordingCoordinator.swift

```diff
@@ -46,30 +54,61 @@ internal class RecordingCoordinator {
         srContextPublisher.setHasReplay(false)
 
         scheduler.schedule { [weak self] in self?.captureNextRecord() }
-        scheduler.start()
 
+        // Start recording immediately if specified.
+        if startRecordingImmediately {
+            startRecording()
+        }
+
+        // Observe changes in the RUM context.
         rumContextObserver.observe(on: scheduler.queue) { [weak self] in self?.onRUMContextChanged(rumContext: $0) }
     }
 
+    /// Enables recording based on user request.
+    func startRecording() {
+        recordingEnabled = true
+        evaluateRecordingConditions()
+    }
+
+    /// Disables recording based on user request.
+    func stopRecording() {
+        recordingEnabled = false
+        evaluateRecordingConditions()
+    }
+
+    // MARK: Private
+
+    /// Evaluates whether recording should start or stop based on user request and sampling.
+    private func evaluateRecordingConditions() {
+       if recordingEnabled && isSampled {
```

Comment:
change-request/ This adds a new use of `isSampled` which isn't synchronized through `@ReadWriteLock`. Because `evaluateRecordingConditions()` can be called on any thread (through start / stop recording), this may crash due to data race if `isSampled` is changed from `onRUMContextChanged()` at the same time.

Reply:
That's a good point! I'll make the change.